// bs-ant-design/lib/js/src/Antd_Breadcrumb.js
// Generated by BUCKLESCRIPT VERSION 2.2.3, PLEASE EDIT WITH CARE
'use strict';

var ReasonReact = require("reason-react/lib/js/src/ReasonReact.js");
var Js_null_undefined = require("bs-platform/lib/js/js_null_undefined.js");
var Breadcrumb = require("antd/lib/breadcrumb");

((require("antd/lib/breadcrumb/style")));

function make(routes, params, separator, itemRender, id, className, style) {
  var partial_arg = {
    routes: Js_null_undefined.fromOption(routes),
    params: Js_null_undefined.fromOption(params),
    separator: Js_null_undefined.fromOption(separator),
    itemRender: Js_null_undefined.fromOption(itemRender),
    id: Js_null_undefined.fromOption(id),
    className: Js_null_undefined.fromOption(className),
    style: Js_null_undefined.fromOption(style)
  };
  var partial_arg$1 = Breadcrumb;
  return (function (param) {
      return ReasonReact.wrapJsForReason(partial_arg$1, partial_arg, param);
    });
}

function make$1(separator, href, id, className, style, children) {
  var tmp = { };
  if (separator) {
    tmp.separator = separator[0];
  }
  if (href) {
    tmp.href = href[0];
  }
  if (id) {
    tmp.id = id[0];
  }
  if (className) {
    tmp.className = className[0];
  }
  if (style) {
    tmp.style = style[0];
  }
  return ReasonReact.wrapJsForReason(Breadcrumb.Item, tmp, children);
}

var Item = /* module */[/* make */make$1];

exports.make = make;
exports.Item = Item;
/*  Not a pure module */
