// bs-css/lib/js/example/example.js
// Generated by BUCKLESCRIPT VERSION 2.2.2, PLEASE EDIT WITH CARE
'use strict';

var Css = require("../src/Css.js");
var Test = require("./test.js");
var React = require("react");
var ReactDOMRe = require("reason-react/lib/js/src/reactDOMRe.js");
var ReasonReact = require("reason-react/lib/js/src/reasonReact.js");

Css.$$global("html, body", /* :: */[
      Css.margin(Css.zero),
      /* :: */[
        Css.marginLeft(Css.px(5)),
        /* :: */[
          Css.padding(Css.zero),
          /* [] */0
        ]
      ]
    ]);

function text(prim) {
  return prim;
}

var fadeIn = Css.keyframes(/* :: */[
      /* tuple */[
        0,
        /* :: */[
          Css.opacity(0),
          /* [] */0
        ]
      ],
      /* :: */[
        /* tuple */[
          100,
          /* :: */[
            Css.opacity(1),
            /* [] */0
          ]
        ],
        /* [] */0
      ]
    ]);

var spin = Css.keyframes(/* :: */[
      /* tuple */[
        0,
        /* :: */[
          Css.transform(Css.rotate(Css.deg(0))),
          /* [] */0
        ]
      ],
      /* :: */[
        /* tuple */[
          100,
          /* :: */[
            Css.transform(Css.rotate(Css.deg(360))),
            /* [] */0
          ]
        ],
        /* [] */0
      ]
    ]);

var styles = {
  page: /* :: */[
    Css.boxSizing(Css.borderBox),
    /* :: */[
      Css.background(Css.linearGradient(Css.deg(45), /* :: */[
                /* tuple */[
                  0,
                  Css.red
                ],
                /* :: */[
                  /* tuple */[
                    100,
                    Css.blue
                  ],
                  /* [] */0
                ]
              ])),
      /* :: */[
        Css.width(Css.vw(100)),
        /* :: */[
          Css.height(Css.vh(100)),
          /* [] */0
        ]
      ]
    ]
  ]
};

var component = ReasonReact.statelessComponent("Page");

function make() {
  var newrecord = component.slice();
  newrecord[/* render */9] = (function () {
      return React.createElement("div", undefined, React.createElement("div", {
                      className: Css.style(styles.page)
                    }, Test.tests));
    });
  return newrecord;
}

var Page = /* module */[
  /* styles */styles,
  /* component */component,
  /* make */make
];

ReactDOMRe.renderToElementWithId(ReasonReact.element(/* None */0, /* None */0, make(/* array */[])), "app");

exports.text = text;
exports.fadeIn = fadeIn;
exports.spin = spin;
exports.Page = Page;
/*  Not a pure module */
