// bs-css/lib/js/example/test.js
// Generated by BUCKLESCRIPT VERSION 2.2.2, PLEASE EDIT WITH CARE
'use strict';

var Css = require("../src/Css.js");
var React = require("react");
var ReasonReact = require("reason-react/lib/js/src/reasonReact.js");

function text(prim) {
  return prim;
}

var arialNarrow = Css.fontFace("Arial FontFace Test", /* :: */[
      Css.localUrl("Arial Narrow"),
      /* [] */0
    ], /* Some */[Css.normal], /* None */0, /* () */0);

var spin = Css.keyframes(/* :: */[
      /* tuple */[
        0,
        /* :: */[
          Css.transform(Css.rotate(Css.deg(0))),
          /* [] */0
        ]
      ],
      /* :: */[
        /* tuple */[
          100,
          /* :: */[
            Css.transform(Css.rotate(Css.deg(360))),
            /* [] */0
          ]
        ],
        /* [] */0
      ]
    ]);

var scaleAnimation = Css.keyframes(/* :: */[
      /* tuple */[
        0,
        /* :: */[
          Css.transform(Css.scale(0.3, 0.3)),
          /* [] */0
        ]
      ],
      /* :: */[
        /* tuple */[
          100,
          /* :: */[
            Css.transform(Css.scale(1.0, 1.0)),
            /* [] */0
          ]
        ],
        /* [] */0
      ]
    ]);

var box_000 = Css.background(Css.red);

var box_001 = /* :: */[
  Css.borderBottom(Css.px(5), Css.solid, Css.black),
  /* :: */[
    Css.width(Css.px(50)),
    /* :: */[
      Css.height(Css.px(50)),
      /* :: */[
        Css.margin(Css.px(10)),
        /* [] */0
      ]
    ]
  ]
];

var box = /* :: */[
  box_000,
  box_001
];

var rowLayout_000 = Css.display(Css.flexBox);

var rowLayout_001 = /* :: */[
  Css.flexDirection(Css.row),
  /* :: */[
    Css.flexWrap(Css.wrap),
    /* [] */0
  ]
];

var rowLayout = /* :: */[
  rowLayout_000,
  rowLayout_001
];

var section_000 = Css.selector("& > h1", /* :: */[
      Css.fontFamily(arialNarrow),
      /* :: */[
        Css.fontSize(Css.px(32)),
        /* :: */[
          Css.fontWeight(300),
          /* [] */0
        ]
      ]
    ]);

var section_001 = /* :: */[
  Css.position(Css.relative),
  /* :: */[
    Css.background(Css.hex("f5f5f5")),
    /* :: */[
      Css.margin(Css.px(20)),
      /* :: */[
        Css.padding(Css.px(20)),
        /* :: */[
          Css.boxShadow(/* None */0, /* Some */[Css.px(1)], /* Some */[Css.px(5)], /* None */0, /* None */0, Css.rgba(0, 0, 0, 0.3)),
          /* :: */[
            Css.boxShadows(/* :: */[
                  Css.boxShadow(/* None */0, /* Some */[Css.px(1)], /* Some */[Css.px(5)], /* None */0, /* None */0, Css.rgba(0, 0, 0, 0.3)),
                  /* :: */[
                    Css.boxShadow(/* None */0, /* Some */[Css.px(1)], /* Some */[Css.px(10)], /* None */0, /* Some */[/* true */1], Css.rgba(255, 255, 255, 0.5)),
                    /* [] */0
                  ]
                ]),
            /* [] */0
          ]
        ]
      ]
    ]
  ]
];

var section = /* :: */[
  section_000,
  section_001
];

var component = ReasonReact.statelessComponent("Section");

function make(name, children) {
  var newrecord = component.slice();
  newrecord[/* render */9] = (function () {
      return React.createElement("section", {
                  className: Css.style(section)
                }, React.createElement("h1", undefined, name), ReasonReact.createDomElement("div", {
                      className: Css.style(rowLayout)
                    }, children));
    });
  return newrecord;
}

var Section = /* module */[
  /* component */component,
  /* make */make
];

var tests = React.createElement("div", {
      className: Css.style(/* :: */[
            Css.background(Css.hex("f5f5f5")),
            /* [] */0
          ])
    }, ReasonReact.element(/* None */0, /* None */0, make("angles", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.transform(Css.rotate(Css.deg(45))),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.transform(Css.rotate(Css.rad(3.1415))),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.transform(Css.rotate(Css.grad(50))),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.transform(Css.rotate(Css.turn(1 / 3))),
                          box
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("colors", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.red),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.rgb(255, 0, 0)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.rgba(255, 0, 0, 0.5)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.hsl(255, 100, 50)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.hsla(255, 100, 50, 0.5)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.hex("FF0000")),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.transparent),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.currentColor),
                          /* :: */[
                            Css.color(Css.blue),
                            box
                          ]
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("gradients", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.linearGradient(Css.deg(45), /* :: */[
                                    /* tuple */[
                                      0,
                                      Css.red
                                    ],
                                    /* :: */[
                                      /* tuple */[
                                        100,
                                        Css.blue
                                      ],
                                      /* [] */0
                                    ]
                                  ])),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.repeatingLinearGradient(Css.deg(45), /* :: */[
                                    /* tuple */[
                                      0,
                                      Css.red
                                    ],
                                    /* :: */[
                                      /* tuple */[
                                        10,
                                        Css.blue
                                      ],
                                      /* [] */0
                                    ]
                                  ])),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.radialGradient(/* :: */[
                                    /* tuple */[
                                      0,
                                      Css.red
                                    ],
                                    /* :: */[
                                      /* tuple */[
                                        100,
                                        Css.blue
                                      ],
                                      /* [] */0
                                    ]
                                  ])),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.repeatingRadialGradient(/* :: */[
                                    /* tuple */[
                                      0,
                                      Css.red
                                    ],
                                    /* :: */[
                                      /* tuple */[
                                        10,
                                        Css.blue
                                      ],
                                      /* [] */0
                                    ]
                                  ])),
                          box
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("lengths", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.ch(1.2)),
                          /* :: */[
                            Css.width(Css.px(10)),
                            /* :: */[
                              Css.maxHeight(Css.pct(50)),
                              /* :: */[
                                Css.maxWidth(Css.pct(100)),
                                box
                              ]
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.cm(1)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.em(1.2)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.ex(1.2)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.mm(10)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.pct(50)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.pt(14)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.px(20)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.rem(2.0)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.vh(1)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.vw(1.0)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.vmax(1.0)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.vmin(1.0)),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.zero),
                          box
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("calc", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.Calc[/* - */0](Css.pt(14), Css.px(10))),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.height(Css.Calc[/* + */1](Css.cm(0.2), Css.mm(10))),
                          box
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("display", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.display(Css.block),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.display(Css.inline),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.display(Css.inlineBlock),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.display(Css.none),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.display(Css.flexBox),
                          box
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("position", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.position(Css.absolute),
                          /* :: */[
                            Css.top(Css.zero),
                            /* :: */[
                              Css.left(Css.zero),
                              /* :: */[
                                Css.right(Css.zero),
                                /* :: */[
                                  Css.bottom(Css.zero),
                                  box
                                ]
                              ]
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.position(Css.relative),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.position(Css.fixed),
                          /* :: */[
                            Css.bottom(Css.px(10)),
                            /* :: */[
                              Css.right(Css.px(10)),
                              box
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.position(Css.$$static),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.position(Css.sticky),
                          box
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("Padding & Margin", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.padding(Css.px(10)),
                          /* :: */[
                            Css.margin(Css.px(10)),
                            box
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.paddingLeft(Css.px(10)),
                          /* :: */[
                            Css.paddingRight(Css.px(10)),
                            /* :: */[
                              Css.paddingTop(Css.px(10)),
                              /* :: */[
                                Css.paddingBottom(Css.px(10)),
                                /* :: */[
                                  Css.marginLeft(Css.px(10)),
                                  /* :: */[
                                    Css.marginRight(Css.px(10)),
                                    /* :: */[
                                      Css.marginTop(Css.px(10)),
                                      /* :: */[
                                        Css.marginBottom(Css.px(10)),
                                        box
                                      ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.padding2(Css.px(10), Css.px(20)),
                          /* :: */[
                            Css.margin2(Css.px(10), Css.px(20)),
                            box
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.padding3(Css.px(10), Css.px(20), Css.px(1)),
                          /* :: */[
                            Css.margin3(Css.px(10), Css.px(20), Css.px(2)),
                            box
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.padding4(Css.px(10), Css.px(15), Css.px(1), Css.px(5)),
                          /* :: */[
                            Css.margin4(Css.px(10), Css.px(15), Css.px(1), Css.px(5)),
                            box
                          ]
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("grid", /* array */[React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.width(Css.pct(100)),
                          /* :: */[
                            Css.height(Css.px(500)),
                            /* :: */[
                              Css.display(Css.grid),
                              /* :: */[
                                Css.gridTemplateColumns(/* :: */[
                                      Css.px(150),
                                      /* :: */[
                                        Css.auto,
                                        /* :: */[
                                          Css.px(150),
                                          /* [] */0
                                        ]
                                      ]
                                    ]),
                                /* :: */[
                                  Css.gridTemplateRows(/* :: */[
                                        Css.px(60),
                                        /* :: */[
                                          Css.auto,
                                          /* [] */0
                                        ]
                                      ]),
                                  /* [] */0
                                ]
                              ]
                            ]
                          ]
                        ])
                  }, React.createElement("div", {
                        className: Css.style(/* :: */[
                              Css.gridColumnStart(1),
                              /* :: */[
                                Css.gridColumnEnd(4),
                                /* :: */[
                                  Css.background(Css.red),
                                  /* :: */[
                                    Css.gridRowStart(1),
                                    /* :: */[
                                      Css.gridRowEnd(1),
                                      /* [] */0
                                    ]
                                  ]
                                ]
                              ]
                            ])
                      }), React.createElement("div", {
                        className: Css.style(/* :: */[
                              Css.background(Css.blue),
                              /* :: */[
                                Css.gridColumn(1, 1),
                                /* :: */[
                                  Css.gridRow(2, 2),
                                  /* [] */0
                                ]
                              ]
                            ])
                      }), React.createElement("div", {
                        className: Css.style(/* :: */[
                              Css.background(Css.green),
                              /* :: */[
                                Css.gridColumn(2, 2),
                                /* :: */[
                                  Css.gridRow(2, 2),
                                  /* :: */[
                                    Css.display(Css.inlineGrid),
                                    /* :: */[
                                      Css.gridTemplateColumns(/* :: */[
                                            Css.px(50),
                                            /* :: */[
                                              Css.auto,
                                              /* [] */0
                                            ]
                                          ]),
                                      /* :: */[
                                        Css.gridTemplateRows(/* :: */[
                                              Css.px(40),
                                              /* :: */[
                                                Css.auto,
                                                /* [] */0
                                              ]
                                            ]),
                                        /* [] */0
                                      ]
                                    ]
                                  ]
                                ]
                              ]
                            ])
                      }, React.createElement("div", {
                            className: Css.style(/* :: */[
                                  Css.background(Css.yellow),
                                  /* :: */[
                                    Css.gridRow(1, 1),
                                    /* :: */[
                                      Css.gridColumn(2, 2),
                                      /* [] */0
                                    ]
                                  ]
                                ])
                          }), React.createElement("div", {
                            className: Css.style(/* :: */[
                                  Css.background(Css.green),
                                  /* :: */[
                                    Css.gridRow(1, 2),
                                    /* :: */[
                                      Css.gridColumn(1, 1),
                                      /* [] */0
                                    ]
                                  ]
                                ])
                          }), React.createElement("div", {
                            className: Css.style(/* :: */[
                                  Css.background(Css.purple),
                                  /* :: */[
                                    Css.gridRow(2, 2),
                                    /* :: */[
                                      Css.gridColumn(2, 2),
                                      /* [] */0
                                    ]
                                  ]
                                ])
                          })), React.createElement("div", {
                        className: Css.style(/* :: */[
                              Css.gridColumnStart(3),
                              /* :: */[
                                Css.gridColumnEnd(3),
                                /* :: */[
                                  Css.background(Css.blue),
                                  /* :: */[
                                    Css.gridRowStart(2),
                                    /* :: */[
                                      Css.gridRowEnd(2),
                                      /* [] */0
                                    ]
                                  ]
                                ]
                              ]
                            ])
                      }))])), ReasonReact.element(/* None */0, /* None */0, make("flexbox", /* array */[React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.flexDirection(Css.column),
                          /* :: */[
                            Css.flexGrow(1),
                            /* :: */[
                              Css.alignItems(Css.stretch),
                              /* :: */[
                                Css.selector("& > *", /* :: */[
                                      Css.marginBottom(Css.px(10)),
                                      /* :: */[
                                        Css.width(Css.pct(100)),
                                        /* [] */0
                                      ]
                                    ]),
                                /* [] */0
                              ]
                            ]
                          ]
                        ])
                  }, React.createElement("div", {
                        className: Css.style(/* :: */[
                              Css.display(Css.flexBox),
                              /* :: */[
                                Css.flexDirection(Css.row),
                                /* :: */[
                                  Css.background(Css.gray),
                                  /* :: */[
                                    Css.alignItems(Css.flexStart),
                                    /* :: */[
                                      Css.justifyContent(Css.flexEnd),
                                      /* [] */0
                                    ]
                                  ]
                                ]
                              ]
                            ])
                      }, React.createElement("div", {
                            className: Css.style(/* :: */[
                                  Css.order(1),
                                  /* :: */[
                                    Css.flexGrow(1),
                                    /* :: */[
                                      Css.flexShrink(1),
                                      /* :: */[
                                        Css.flexBasis(Css.auto),
                                        box
                                      ]
                                    ]
                                  ]
                                ])
                          }), React.createElement("div", {
                            className: Css.style(/* :: */[
                                  Css.alignSelf(Css.flexEnd),
                                  box
                                ])
                          }), React.createElement("div", {
                            className: Css.style(box)
                          })), React.createElement("div", {
                        className: Css.style(/* :: */[
                              Css.display(Css.flexBox),
                              /* :: */[
                                Css.flexDirection(Css.column),
                                /* :: */[
                                  Css.background(Css.gray),
                                  /* :: */[
                                    Css.alignItems(Css.baseline),
                                    /* :: */[
                                      Css.justifyContent(Css.flexStart),
                                      /* [] */0
                                    ]
                                  ]
                                ]
                              ]
                            ])
                      }, React.createElement("div", {
                            className: Css.style(box)
                          }), React.createElement("div", {
                            className: Css.style(box)
                          }), React.createElement("div", {
                            className: Css.style(box)
                          })), React.createElement("div", {
                        className: Css.style(/* :: */[
                              Css.display(Css.flexBox),
                              /* :: */[
                                Css.flexDirection(Css.rowReverse),
                                /* :: */[
                                  Css.background(Css.gray),
                                  /* :: */[
                                    Css.alignItems(Css.center),
                                    /* :: */[
                                      Css.justifyContent(Css.spaceBetween),
                                      /* [] */0
                                    ]
                                  ]
                                ]
                              ]
                            ])
                      }, React.createElement("div", {
                            className: Css.style(box)
                          }), React.createElement("div", {
                            className: Css.style(/* :: */[
                                  Css.height(Css.px(50)),
                                  /* :: */[
                                    Css.width(Css.px(50)),
                                    box
                                  ]
                                ])
                          }), React.createElement("div", {
                            className: Css.style(box)
                          })), React.createElement("div", {
                        className: Css.style(/* :: */[
                              Css.display(Css.flexBox),
                              /* :: */[
                                Css.flexDirection(Css.columnReverse),
                                /* :: */[
                                  Css.background(Css.gray),
                                  /* :: */[
                                    Css.alignItems(Css.flexEnd),
                                    /* :: */[
                                      Css.justifyContent(Css.flexEnd),
                                      /* [] */0
                                    ]
                                  ]
                                ]
                              ]
                            ])
                      }, React.createElement("div", {
                            className: Css.style(box)
                          }), React.createElement("div", {
                            className: Css.style(/* :: */[
                                  Css.height(Css.px(50)),
                                  /* :: */[
                                    Css.width(Css.px(50)),
                                    box
                                  ]
                                ])
                          }), React.createElement("div", {
                            className: Css.style(box)
                          })), React.createElement("div", {
                        className: Css.style(/* :: */[
                              Css.display(Css.flexBox),
                              /* :: */[
                                Css.flexDirection(Css.row),
                                /* :: */[
                                  Css.background(Css.gray),
                                  /* :: */[
                                    Css.alignItems(Css.stretch),
                                    /* :: */[
                                      Css.justifyContent(Css.spaceAround),
                                      /* [] */0
                                    ]
                                  ]
                                ]
                              ]
                            ])
                      }, React.createElement("div", {
                            className: Css.style(box)
                          }), React.createElement("div", {
                            className: Css.style(/* :: */[
                                  Css.height(Css.px(50)),
                                  /* :: */[
                                    Css.width(Css.px(50)),
                                    box
                                  ]
                                ])
                          }), React.createElement("div", {
                            className: Css.style(box)
                          })))])), ReasonReact.element(/* None */0, /* None */0, make("float", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.$$float(/* left */-944764921),
                          /* :: */[
                            Css.clear(/* right */-379319332),
                            box
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.$$float(/* right */-379319332),
                          /* :: */[
                            Css.clear(/* left */-944764921),
                            box
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.$$float(Css.none),
                          /* :: */[
                            Css.clear(Css.both),
                            box
                          ]
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("overflow", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.overflow(Css.hidden),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.overflow(Css.visible),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.overflow(Css.auto),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.overflow(Css.scroll),
                          box
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("border", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.border(Css.px(5), Css.solid, Css.blue),
                          /* :: */[
                            Css.borderRadius(Css.px(1000)),
                            box
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.border(Css.px(5), Css.solid, Css.green),
                          /* :: */[
                            Css.borderTopRightRadius(Css.px(1000)),
                            /* :: */[
                              Css.borderTopLeftRadius(Css.px(1000)),
                              /* :: */[
                                Css.borderBottomRightRadius(Css.px(1000)),
                                /* :: */[
                                  Css.borderBottomLeftRadius(Css.px(1000)),
                                  box
                                ]
                              ]
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.borderTop(Css.px(5), Css.dashed, Css.hex("FFF")),
                          /* :: */[
                            Css.borderRight(Css.px(5), Css.dotted, Css.rgb(0, 0, 0)),
                            /* :: */[
                              Css.borderBottom(Css.px(5), Css.none, Css.green),
                              /* :: */[
                                Css.borderLeft(Css.px(5), Css.solid, Css.blue),
                                box
                              ]
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.borderWidth(Css.px(5)),
                          /* :: */[
                            Css.borderStyle(Css.solid),
                            /* :: */[
                              Css.borderColor(Css.blue),
                              box
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.borderTopWidth(Css.px(5)),
                          /* :: */[
                            Css.borderTopStyle(Css.solid),
                            /* :: */[
                              Css.borderTopColor(Css.blue),
                              box
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.borderBottomWidth(Css.px(5)),
                          /* :: */[
                            Css.borderBottomStyle(Css.solid),
                            /* :: */[
                              Css.borderBottomColor(Css.blue),
                              box
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.borderLeftWidth(Css.px(5)),
                          /* :: */[
                            Css.borderLeftStyle(Css.solid),
                            /* :: */[
                              Css.borderLeftColor(Css.blue),
                              box
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.borderRightWidth(Css.px(5)),
                          /* :: */[
                            Css.borderRightStyle(Css.solid),
                            /* :: */[
                              Css.borderRightColor(Css.blue),
                              box
                            ]
                          ]
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("background", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.red),
                          /* :: */[
                            Css.backgroundAttachment(Css.scroll),
                            /* :: */[
                              Css.backgroundClip(Css.borderBox),
                              /* :: */[
                                Css.backgroundOrigin(Css.borderBox),
                                /* :: */[
                                  Css.backgroundPosition(Css.pct(50), Css.pct(50)),
                                  /* :: */[
                                    Css.backgroundRepeat(Css.noRepeat),
                                    /* :: */[
                                      Css.backgroundSize(Css.size(Css.px(100), Css.px(100))),
                                      box
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.radialGradient(/* :: */[
                                    /* tuple */[
                                      0,
                                      Css.red
                                    ],
                                    /* :: */[
                                      /* tuple */[
                                        10,
                                        Css.blue
                                      ],
                                      /* [] */0
                                    ]
                                  ])),
                          /* :: */[
                            Css.backgroundAttachment(Css.fixed),
                            /* :: */[
                              Css.backgroundClip(Css.contentBox),
                              /* :: */[
                                Css.backgroundOrigin(Css.contentBox),
                                /* :: */[
                                  Css.backgroundRepeat(Css.repeat),
                                  /* :: */[
                                    Css.backgroundSize(Css.auto),
                                    box
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.background(Css.url("./img-29.jpg")),
                          /* :: */[
                            Css.backgroundAttachment(Css.local),
                            /* :: */[
                              Css.backgroundClip(Css.paddingBox),
                              /* :: */[
                                Css.backgroundOrigin(Css.paddingBox),
                                /* :: */[
                                  Css.backgroundRepeat(Css.repeatX),
                                  /* :: */[
                                    Css.backgroundSize(Css.cover),
                                    box
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.backgroundColor(Css.rgb(0, 0, 255)),
                          /* :: */[
                            Css.backgroundImage(Css.linearGradient(Css.deg(45), /* :: */[
                                      /* tuple */[
                                        0,
                                        Css.green
                                      ],
                                      /* :: */[
                                        /* tuple */[
                                          50,
                                          Css.red
                                        ],
                                        /* :: */[
                                          /* tuple */[
                                            100,
                                            Css.yellow
                                          ],
                                          /* [] */0
                                        ]
                                      ]
                                    ])),
                            /* :: */[
                              Css.backgroundRepeat(Css.repeatY),
                              /* :: */[
                                Css.backgroundSize(Css.contain),
                                box
                              ]
                            ]
                          ]
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("cursor", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* pointer */-786317123),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* alias */563171728),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* allScroll */365450254),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* auto */-1065951377),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* cell */-1044569694),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* contextMenu */626862894),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* default */465819841),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* none */-922086728),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* crosshair */180897442),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* copy */-1044071499),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* grab */-999567412),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* grabbing */-19620980),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* help */-989121855),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* move */-933174511),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* notAllowed */939907157),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* progress */103479213),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* text */-856044371),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* wait */-822977931),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* zoomIn */-88732200),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.cursor(/* zoomOut */-459627717),
                          box
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("list", /* array */[React.createElement("ul", undefined, React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyle(/* disc */-1033279659, Css.inside, Css.none),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* circle */-703761904),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* square */-655228771),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* decimal */-422333295),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* lowerAlpha */-571131491),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* upperAlpha */-484197732),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* lowerGreek */-699686657),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* lowerLatin */739986499),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* upperLatin */826920258),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* lowerRoman */700345660),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* upperRoman */787279419),
                              /* [] */0
                            ])
                      }), React.createElement("li", {
                        className: Css.style(/* :: */[
                              Css.listStyleType(/* disc */-1033279659),
                              /* :: */[
                                Css.listStylePosition(Css.inside),
                                /* :: */[
                                  Css.listStyleImage(Css.url("./facebook.png")),
                                  /* [] */0
                                ]
                              ]
                            ])
                      }))])), ReasonReact.element(/* None */0, /* None */0, make("outline", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.outline(Css.px(5), /* double */852175633, Css.green),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.outlineStyle(Css.solid),
                          /* :: */[
                            Css.outlineWidth(Css.px(5)),
                            /* :: */[
                              Css.outlineColor(Css.green),
                              /* :: */[
                                Css.outlineOffset(Css.px(5)),
                                box
                              ]
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.outline(Css.px(5), /* double */852175633, Css.red),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.outline(Css.px(5), /* ridge */-379468757, Css.red),
                          box
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("transform", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.opacity(0.5),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.perspective(Css.px(500)),
                          /* :: */[
                            Css.transform(Css.rotate(Css.deg(10))),
                            box
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.transforms(/* :: */[
                                Css.translate(Css.px(10), Css.pct(10)),
                                /* :: */[
                                  Css.skew(Css.deg(10), Css.deg(10)),
                                  /* [] */0
                                ]
                              ]),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.transform(Css.rotate(Css.deg(19))),
                          /* :: */[
                            Css.transformOrigin(Css.pct(50), Css.pct(50)),
                            /* :: */[
                              Css.transformStyle(/* preserve3d */589702045),
                              /* :: */[
                                Css.perspective(Css.px(900)),
                                /* :: */[
                                  Css.perspectiveOrigin(Css.pct(10), Css.pct(10)),
                                  box
                                ]
                              ]
                            ]
                          ]
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.transform(Css.translate(Css.px(10), Css.pct(10))),
                          /* :: */[
                            Css.transformOrigin3d(Css.px(10), Css.px(10), Css.px(10)),
                            box
                          ]
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("transition", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.transition(/* Some */[300], /* Some */[300], /* Some */[Css.easeInOut], "transform"),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.transitionProperty("height"),
                          /* :: */[
                            Css.transitionDelay(300),
                            /* :: */[
                              Css.transitionDuration(300),
                              /* :: */[
                                Css.transitionTimingFunction(Css.linear),
                                box
                              ]
                            ]
                          ]
                        ])
                  })
            ])), ReasonReact.element(/* None */0, /* None */0, make("text", /* array */[React.createElement("p", {
                    className: Css.style(/* :: */[
                          Css.color(Css.black),
                          /* :: */[
                            Css.fontFamily("Helvetica, sans-serif"),
                            /* :: */[
                              Css.fontSize(Css.pt(18)),
                              /* :: */[
                                Css.fontVariant(/* smallCaps */-386684792),
                                /* :: */[
                                  Css.fontStyle(Css.italic),
                                  /* :: */[
                                    Css.fontWeight(300),
                                    /* :: */[
                                      Css.letterSpacing(Css.px(3)),
                                      /* :: */[
                                        Css.lineHeight(2),
                                        /* :: */[
                                          Css.textAlign(/* left */-944764921),
                                          /* :: */[
                                            Css.textDecoration(Css.underline),
                                            /* :: */[
                                              Css.textDecorationColor(Css.pink),
                                              /* :: */[
                                                Css.textDecorationStyle(Css.wavy),
                                                /* :: */[
                                                  Css.textIndent(Css.px(10)),
                                                  /* :: */[
                                                    Css.textOverflow(Css.clip),
                                                    /* :: */[
                                                      Css.textShadow(/* None */0, /* Some */[Css.px(3)], /* Some */[Css.px(2)], Css.black),
                                                      /* :: */[
                                                        Css.textTransform(Css.capitalize),
                                                        /* :: */[
                                                          Css.verticalAlign(Css.sub),
                                                          /* :: */[
                                                            Css.whiteSpace(Css.normal),
                                                            /* :: */[
                                                              Css.wordBreak(Css.breakAll),
                                                              /* :: */[
                                                                Css.wordSpacing(Css.px(20)),
                                                                /* :: */[
                                                                  Css.wordWrap(Css.breakWord),
                                                                  /* [] */0
                                                                ]
                                                              ]
                                                            ]
                                                          ]
                                                        ]
                                                      ]
                                                    ]
                                                  ]
                                                ]
                                              ]
                                            ]
                                          ]
                                        ]
                                      ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ])
                  }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.")])), ReasonReact.element(/* None */0, /* None */0, make("animation", /* array */[
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.animation(/* Some */[300], /* Some */[300], /* Some */[Css.reverse], /* Some */[Css.linear], /* Some */[Css.forwards], /* Some */[Css.running], /* Some */[Css.infinite], spin),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.animations(/* :: */[
                                Css.animation(/* Some */[300], /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* Some */[Css.infinite], spin),
                                /* :: */[
                                  Css.animation(/* Some */[300], /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* Some */[Css.infinite], scaleAnimation),
                                  /* [] */0
                                ]
                              ]),
                          box
                        ])
                  }),
              React.createElement("div", {
                    className: Css.style(/* :: */[
                          Css.animationName(spin),
                          /* :: */[
                            Css.animationTimingFunction(Css.easeIn),
                            /* :: */[
                              Css.animationDuration(300),
                              /* :: */[
                                Css.animationDelay(300),
                                /* :: */[
                                  Css.animationDirection(Css.normal),
                                  /* :: */[
                                    Css.animationFillMode(Css.backwards),
                                    /* :: */[
                                      Css.animationPlayState(Css.paused),
                                      /* :: */[
                                        Css.animationIterationCount(Css.count(5)),
                                        box
                                      ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ])
                  })
            ])));

exports.text = text;
exports.arialNarrow = arialNarrow;
exports.spin = spin;
exports.scaleAnimation = scaleAnimation;
exports.box = box;
exports.rowLayout = rowLayout;
exports.section = section;
exports.Section = Section;
exports.tests = tests;
/* arialNarrow Not a pure module */
