// reason-react-primitives/lib/js/src/button.js
// Generated by BUCKLESCRIPT VERSION 2.2.3, PLEASE EDIT WITH CARE
'use strict';

var ReasonReact = require("reason-react/lib/js/src/ReasonReact.js");
var TouchableOpacity = require("./touchableOpacity.js");

var component = ReasonReact.statelessComponent("Button");

function make(onPress, color, title, $staropt$star, $staropt$star$1, _) {
  var disabled = $staropt$star ? $staropt$star[0] : /* false */0;
  var round = $staropt$star$1 ? $staropt$star$1[0] : /* true */1;
  var newrecord = component.slice();
  newrecord[/* render */9] = (function () {
      return ReasonReact.element(/* None */0, /* None */0, TouchableOpacity.make(onPress, /* Some */[{
                        backgroundColor: color,
                        color: "#fff",
                        display: "inline-flex",
                        fontSize: "14px",
                        padding: "8px",
                        borderRadius: round !== 0 ? "3px" : ""
                      }], /* Some */[{
                        backgroundImage: "linear-gradient(to bottom, rgba(0, 0, 0, 0.1), rgba(0, 0, 0, 0.1))",
                        outline: "none"
                      }], /* None */0, /* Some */[disabled], /* Some */[{
                        cursor: "default",
                        outline: "none",
                        opacity: "0.4"
                      }], /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* array */[title]));
    });
  return newrecord;
}

exports.component = component;
exports.make = make;
/* component Not a pure module */
