// reason-react-primitives/lib/js/src/buttonGroup.js
// Generated by BUCKLESCRIPT VERSION 2.2.3, PLEASE EDIT WITH CARE
'use strict';

var $$Array = require("bs-platform/lib/js/array.js");
var React = require("react");
var ReactDOMRe = require("reason-react/lib/js/src/ReactDOMRe.js");
var ReasonReact = require("reason-react/lib/js/src/ReasonReact.js");

var component = ReasonReact.statelessComponent("Button");

var buttonGroupStyle = {
  display: "flex",
  overflow: "hidden",
  borderRadius: "3px",
  alignItems: "center",
  flexDirection: "row"
};

function addKey(index, item) {
  return React.createElement("div", {
              key: String(index)
            }, item);
}

function make(style, children) {
  var newrecord = component.slice();
  newrecord[/* render */9] = (function () {
      return React.createElement("div", {
                  style: style ? ReactDOMRe.Style[/* combine */0](buttonGroupStyle, style[0]) : buttonGroupStyle
                }, $$Array.mapi(addKey, children));
    });
  return newrecord;
}

exports.component = component;
exports.buttonGroupStyle = buttonGroupStyle;
exports.addKey = addKey;
exports.make = make;
/* component Not a pure module */
