// reason-react-primitives/lib/js/src/confirm.js
// Generated by BUCKLESCRIPT VERSION 2.2.3, PLEASE EDIT WITH CARE
'use strict';

var Text = require("./text.js");
var React = require("react");
var Button = require("./button.js");
var ReactDOMRe = require("reason-react/lib/js/src/ReactDOMRe.js");
var ReasonReact = require("reason-react/lib/js/src/ReasonReact.js");

var component = ReasonReact.statelessComponent("Confirm");

var root = {
  display: "flex",
  height: "100vh",
  width: "100vw",
  alignItems: "center",
  flexDirection: "column",
  justifyContent: "center",
  boxSizing: "border-box"
};

var layer = {
  backgroundColor: "rgba(0, 0, 0, 0.3)",
  bottom: "0",
  left: "0",
  position: "fixed",
  right: "0",
  top: "0",
  transition: "background-color .5s ease-in-out"
};

var container = {
  backgroundColor: "#fff",
  margin: "10px",
  overflow: "hidden",
  position: "relative",
  zIndex: "0",
  borderRadius: "4px",
  boxShadow: "0 2px 4px rgba(0, 0, 0, 0.16)"
};

var icon = {
  display: "flex",
  padding: "10px",
  alignItems: "center",
  flexDirection: "row",
  justifyContent: "center"
};

var message = {
  borderBottom: "1px solid rgba(0, 0, 0, 0.1)",
  padding: "20px 10px",
  textAlign: "center"
};

var messageText = {
  color: "rgba(0, 0, 0, 0.4)",
  fontSize: "16px"
};

var buttons = {
  backgroundColor: "#fafafa",
  display: "flex",
  padding: "10px",
  flexDirection: "row",
  justifyContent: "space-between"
};

var Styles = /* module */[
  /* root */root,
  /* layer */layer,
  /* container */container,
  /* icon */icon,
  /* message */message,
  /* messageText */messageText,
  /* buttons */buttons
];

function make(icon$1, message$1, confirmWording, cancelWording, onConfirm, onCancel, maxWidth, _) {
  var newrecord = component.slice();
  newrecord[/* render */9] = (function () {
      return React.createElement("div", {
                  style: root
                }, React.createElement("div", {
                      style: layer
                    }), React.createElement("div", {
                      style: ReactDOMRe.Style[/* combine */0](container, {
                            maxWidth: maxWidth ? String(maxWidth[0]) + "px" : ""
                          })
                    }, icon$1 ? React.createElement("div", {
                            style: icon
                          }, icon$1[0]) : null, React.createElement("div", {
                          style: message
                        }, ReasonReact.element(/* None */0, /* None */0, Text.make(/* Some */[messageText], /* array */[message$1]))), React.createElement("div", {
                          style: buttons
                        }, ReasonReact.element(/* None */0, /* None */0, Button.make(onCancel, "#9B9B9B", cancelWording, /* None */0, /* None */0, /* array */[])), React.createElement("div", {
                              style: {
                                height: "10px",
                                width: "10px",
                                flexShrink: "0"
                              }
                            }), ReasonReact.element(/* None */0, /* None */0, Button.make(onConfirm, "#fb5", confirmWording, /* None */0, /* None */0, /* array */[])))));
    });
  return newrecord;
}

exports.component = component;
exports.Styles = Styles;
exports.make = make;
/* component Not a pure module */
