// reason-react-primitives/lib/js/src/image.js
// Generated by BUCKLESCRIPT VERSION 2.2.3, PLEASE EDIT WITH CARE
'use strict';

var Block = require("bs-platform/lib/js/block.js");
var React = require("react");
var ReasonReact = require("reason-react/lib/js/src/ReasonReact.js");
var ScreenReaderContent = require("./screenReaderContent.js");

var component = ReasonReact.reducerComponent("Image");

function make(source, resizeMode, width, height, backgroundColor, $staropt$star, alternativeText, _) {
  var round = $staropt$star ? $staropt$star[0] : /* false */0;
  var newrecord = component.slice();
  newrecord[/* didMount */4] = (function () {
      return /* Update */Block.__(0, [/* record */[/* isReady : true */1]]);
    });
  newrecord[/* render */9] = (function (param) {
      var match = param[/* state */2][/* isReady */0];
      var tmp;
      switch (resizeMode) {
        case 0 : 
            tmp = "cover";
            break;
        case 1 : 
            tmp = "contain";
            break;
        case 2 : 
            tmp = "auto";
            break;
        
      }
      return React.createElement("div", {
                  style: {
                    backgroundColor: backgroundColor,
                    backgroundImage: match !== 0 ? "url(" + (source + ")") : "",
                    backgroundPosition: "50% 50%",
                    backgroundRepeat: "no-repeat",
                    height: String(height) + "px",
                    width: String(width) + "px",
                    backgroundSize: tmp,
                    borderRadius: round !== 0 ? String(width < height ? width : height) + "px" : "0px"
                  }
                }, alternativeText ? ReasonReact.element(/* None */0, /* None */0, ScreenReaderContent.make(alternativeText[0], /* array */[])) : null);
    });
  newrecord[/* initialState */10] = (function () {
      return /* record */[/* isReady : false */0];
    });
  newrecord[/* reducer */12] = (function (_, _$1) {
      return /* NoUpdate */0;
    });
  return newrecord;
}

exports.component = component;
exports.make = make;
/* component Not a pure module */
