// reason-react-primitives/lib/js/src/primitives.js
// Generated by BUCKLESCRIPT VERSION 2.2.3, PLEASE EDIT WITH CARE
'use strict';


var ActivityIndicator = 0;

var Button = 0;

var ButtonGroup = 0;

var Dialog = 0;

var FixedCollectionView = 0;

var HorizontalSeparator = 0;

var Image = 0;

var LayerManager = 0;

var ListView = 0;

var Picker = 0;

var ScrollView = 0;

var SectionedListView = 0;

var Sprite = 0;

var Switch = 0;

var TabbedView = 0;

var Text = 0;

var TextInput = 0;

var Tooltip = 0;

var TouchableHighlight = 0;

var TouchableOpacity = 0;

var View = 0;

exports.ActivityIndicator = ActivityIndicator;
exports.Button = Button;
exports.ButtonGroup = ButtonGroup;
exports.Dialog = Dialog;
exports.FixedCollectionView = FixedCollectionView;
exports.HorizontalSeparator = HorizontalSeparator;
exports.Image = Image;
exports.LayerManager = LayerManager;
exports.ListView = ListView;
exports.Picker = Picker;
exports.ScrollView = ScrollView;
exports.SectionedListView = SectionedListView;
exports.Sprite = Sprite;
exports.Switch = Switch;
exports.TabbedView = TabbedView;
exports.Text = Text;
exports.TextInput = TextInput;
exports.Tooltip = Tooltip;
exports.TouchableHighlight = TouchableHighlight;
exports.TouchableOpacity = TouchableOpacity;
exports.View = View;
/* No side effect */
