// reason-react-primitives/lib/js/src/sectionedListView.js
// Generated by BUCKLESCRIPT VERSION 2.2.3, PLEASE EDIT WITH CARE
'use strict';

var List = require("bs-platform/lib/js/list.js");
var $$Array = require("bs-platform/lib/js/array.js");
var React = require("react");
var ReasonReact = require("reason-react/lib/js/src/ReasonReact.js");

var component = ReasonReact.statelessComponent("SectionedListView");

var container = {
  width: "100%",
  flexGrow: "1"
};

var sectionTitle = {
  borderBottom: "1px solid rgba(0, 0, 0, 0.1)",
  color: "rgba(0, 0, 0, 0.5)",
  fontSize: "12px",
  letterSpacing: "1px",
  lineHeight: "1",
  padding: "10px",
  textTransform: "uppercase"
};

var Styles = /* module */[
  /* container */container,
  /* sectionTitle */sectionTitle
];

function addKey(index, item) {
  return React.createElement("div", {
              key: String(index)
            }, item);
}

function renderSection(backgroundColor, sectionBackgroundColor, index, param) {
  return React.createElement("div", {
              key: String(index),
              style: {
                backgroundColor: backgroundColor,
                paddingBottom: "10px"
              }
            }, React.createElement("div", {
                  style: sectionTitle
                }, param[0]), React.createElement("div", {
                  style: {
                    backgroundColor: sectionBackgroundColor
                  }
                }, $$Array.of_list(List.mapi(addKey, param[1]))));
}

function make(sections, $staropt$star, $staropt$star$1, _) {
  var backgroundColor = $staropt$star ? $staropt$star[0] : "rgb(248, 248, 248)";
  var sectionBackgroundColor = $staropt$star$1 ? $staropt$star$1[0] : "#fff";
  var newrecord = component.slice();
  newrecord[/* render */9] = (function () {
      return React.createElement("div", {
                  style: container
                }, $$Array.of_list(List.mapi((function (param, param$1) {
                            return renderSection(backgroundColor, sectionBackgroundColor, param, param$1);
                          }), sections)));
    });
  return newrecord;
}

exports.component = component;
exports.Styles = Styles;
exports.addKey = addKey;
exports.renderSection = renderSection;
exports.make = make;
/* component Not a pure module */
