// reason-react-primitives/lib/js/src/tag.js
// Generated by BUCKLESCRIPT VERSION 2.2.3, PLEASE EDIT WITH CARE
'use strict';

var React = require("react");
var Pervasives = require("bs-platform/lib/js/pervasives.js");
var ReactDOMRe = require("reason-react/lib/js/src/ReactDOMRe.js");
var ReasonReact = require("reason-react/lib/js/src/ReasonReact.js");
var TouchableOpacity = require("./touchableOpacity.js");

var component = ReasonReact.statelessComponent("Tag");

function tupleToColor(param, $staropt$star, _) {
  var alpha = $staropt$star ? $staropt$star[0] : 1.0;
  var match = alpha !== 1.0 ? /* tuple */[
      "rgba(",
      "," + (Pervasives.string_of_float(alpha) + ")")
    ] : /* tuple */[
      "rgb(",
      ")"
    ];
  return match[0] + (String(param[0]) + ("," + (String(param[1]) + ("," + (String(param[2]) + match[1])))));
}

var root = {
  display: "inline-flex",
  height: "30px",
  maxWidth: "100%",
  paddingRight: "7px",
  borderRadius: "24px",
  alignItems: "center",
  boxSizing: "borderBox"
};

var name = {
  fontSize: "12px",
  overflow: "hidden",
  padding: "0 10px",
  whiteSpace: "nowrap",
  textOverflow: "ellipsis"
};

var button = {
  color: "#fff",
  fontSize: "12px",
  height: "18px",
  lineHeight: "18px",
  textAlign: "center",
  width: "18px",
  borderRadius: "9px",
  flexShrink: "0"
};

var Styles = /* module */[
  /* root */root,
  /* name */name,
  /* button */button
];

function make(name$1, $staropt$star, onPress, _) {
  var color = $staropt$star ? $staropt$star[0] : /* tuple */[
      74,
      144,
      226
    ];
  var solidColor = tupleToColor(color, /* None */0, /* () */0);
  var washedColor = tupleToColor(color, /* Some */[0.17], /* () */0);
  var newrecord = component.slice();
  newrecord[/* render */9] = (function () {
      return React.createElement("div", {
                  style: ReactDOMRe.Style[/* combine */0](root, {
                        backgroundColor: washedColor,
                        border: "1px solid " + solidColor
                      })
                }, React.createElement("span", {
                      style: ReactDOMRe.Style[/* combine */0](name, {
                            color: solidColor
                          })
                    }, name$1), ReasonReact.element(/* None */0, /* None */0, TouchableOpacity.make(onPress, /* Some */[ReactDOMRe.Style[/* combine */0](button, {
                                backgroundColor: solidColor
                              })], /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* array */["✕"])));
    });
  return newrecord;
}

exports.component = component;
exports.tupleToColor = tupleToColor;
exports.Styles = Styles;
exports.make = make;
/* component Not a pure module */
